Easydict/NewApp/Utility/Extensions/QueryService+ConfigurableService/OpenAIService+ConfigurableService.swift

```diff
@@ -9,27 +9,101 @@
+@available(macOS 13.0, *)
 extension EZOpenAIService: ConfigurableService {
     func configurationListItems() -> some View {
-        ServiceStringConfigurationSection(
-            textFieldTitleKey: "service.configuration.openai.api_key.header",
-            headerTitleKey: "service.configuration.openai.api_key.title",
-            key: .openAIAPIKey,
-            prompt: "service.configuration.openai.api_key.prompt",
-            footer: {
-                Text("service.configuration.openai.api_key.footer")
-            }
-        )
-
-        ServiceStringConfigurationSection(
-            textFieldTitleKey: "service.configuration.openai.translation.header",
-            headerTitleKey: "service.configuration.openai.translation.title",
-            key: .openAITranslation,
-            prompt: "service.configuration.openai.translation.prompt",
-            footer: {
-                Text("service.configuration.openai.translation.footer")
-            }
-        )
+        ServiceConfigurationSecretSectionView(service: self, observeKeys: [.openAIAPIKey]) {
+            ServiceConfigurationSecureInputCell(
+                textFieldTitleKey: "service.configuration.openai.api_key.title",
+                key: .openAIAPIKey,
+                placeholder: "service.configuration.openai.api_key.placeholder"
+            )
+            // endpoint
+            ServiceConfigurationInputCell(
+                textFieldTitleKey: "service.configuration.openai.endpoint.title",
+                key: .openAIEndPoint,
+                placeholder: "service.configuration.openai.endpoint.placeholder"
+            )
+            // model
+            ServiceConfigurationPickerCell(
+                titleKey: "service.configuration.openai.model.title",
+                key: .openAIModel,
+                values: OpenAIModels.allCases
+            )
+
+            ServiceConfigurationToggleCell(
+                titleKey: "service.configuration.openai.translation.title",
+                key: .openAITranslation
+            )
+            ServiceConfigurationToggleCell(
+                titleKey: "service.configuration.openai.sentence.title",
+                key: .openAISentence
+            )
+            ServiceConfigurationToggleCell(
+                titleKey: "service.configuration.openai.dictionary.title",
+                key: .openAIDictionary
+            )
+            ServiceConfigurationPickerCell(
+                titleKey: "service.configuration.openai.usage_status.title",
+                key: .openAIServiceUsageStatus,
+                values: OpenAIUsageStats.allCases
+            )
+        }
+    }
+}
+
+protocol EnumLocalizedStringConvertible {
+    var title: String { get }
+}
+
+enum OpenAIModels: String, CaseIterable, Identifiable {
+    var id: Self {
+        self
+    }
+
+    case gpt3_5_turbo = "gpt-3.5-turbo"
+    case gpt3_5_turbo_1106 = "gpt-3.5-turbo-1106"
+    case gpt4
+    case gpt4_32k = "gpt-4-32k"
```

Comment:
We need to update to use the latest and cheapest models, like `gpt-3.5-turbo-0125` and `
gpt-3.5-turbo-0125`

https://platform.openai.com/docs/models/gpt-4-and-gpt-4-turbo

> gpt-3.5-turbo-0125 is the flagship model of this family, supports a 16K context window and is optimized for dialog.

<img width="851" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/74457f5b-259f-4db3-abec-fec3e9eca109">

<img width="848" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/659351db-e3c3-48ad-b7b3-e5a25bfcf8dd">

Reply:
<img width="825" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/80435ef1-40e1-482a-a82a-515847541333">

Reply:
<img width="825" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/95ed6030-07e5-4b58-8a17-0bc5d37cb468">

Reply:
ok, done.